demo/api/src/importer/pipes/parsers/csv-data-transformer.pipe.ts

```diff
@@ -0,0 +1,208 @@
+    } {
+        let transformedValue: PossibleValueTypes = value;
+        const valueMappingHasEmptyString =
+            field.valueMapping &&
+            Object.keys(field.valueMapping).find((key: string) => {
+                return key === "";
+            }) !== undefined;
+        // Skip empty values if valueMapping contains a value for an empty string key for CSV
+        if (transformedValue === "" && !valueMappingHasEmptyString) {
+            if (isNullable(propertyMetadata)) {
+                transformedValue = null;
+            } else {
+                return {
+                    error: {
+                        property: field.key,
+                        name: `${field.csvColumnName}`.toUpperCase(),
+                        errorMessage: "Non-nullable field is empty",
+                        value,
+                    },
+                };
+            }
+        } else if (columnIsBoolean(propertyMetadata) && field.valueMapping) {
+            transformedValue = field.valueMapping[value];
+        } else if (columnIsInteger(propertyMetadata)) {
+            transformedValue = parseInt(value);
+            if (isNaN(transformedValue)) {
+                return {
+                    error: {
+                        property: field.key,
+                        name: `${field.csvColumnName}`.toUpperCase(),
+                        errorMessage: "Integer field is NaN",
+                        value,
+                    },
+                };
+            }
+        } else if (columnIsFloat(propertyMetadata)) {
+            transformedValue = parseFloat(transformedValue);
+            if (isNaN(transformedValue)) {
+                return {
+                    error: {
+                        property: field.key,
+                        name: `${field.csvColumnName}`.toUpperCase(),
+                        errorMessage: "Float field is NaN",
+                        value,
+                    },
+                };
+            }
+        } else if (columnIsDate(propertyMetadata)) {
+            let date: Date;
+            if (field.dateFormatString) {
+                date = parse(transformedValue, field.dateFormatString, new Date());
+            } else if (this.options.entityDateFormatString) {
+                date = parse(transformedValue, this.options.entityDateFormatString, new Date());
+            } else {
+                date = parseISO(transformedValue);
+            }
+            if (!(date instanceof Date && !isNaN(date.valueOf()))) {
+                return {
+                    error: {
+                        property: field.key,
+                        name: `${field.csvColumnName}`.toUpperCase(),
+                        errorMessage: `Invalid Date: ${transformedValue}`,
+                        value: transformedValue,
+                    },
+                };
+            }
+            transformedValue = date;
+        }
+        if (field.key === "soldCount") {
```

Comment:
this does probably not belong to this generic class. if this debug-log stuff would be handy, please use some kind of `debug: true` flag or log-level-config instead.